modules/catalog-compaction/README.md

```diff
@@ -1 +1,101 @@
 # Catalog compaction module
+
+> **_NOTE_** Compaction has been moved to a separate module to eliminate circular dependencies,
+as it requires some components that may themselves depend on the catalog module. Please 
+refer to the catalog's module [readme](../catalog/README.md) to see more details about 
+catalog service and update log.
+
+## Overview
+
+During schema changes, the catalog update log stores incremental updates. Each update
+increases the catalog version. Over time, the log may grow to a humongous  size. To 
+address this, snapshotting was introduced to UpdateLog. Snapshotting means replacing 
+incremental updates with a snapshot.
+
+But different components can refer to a specific version of the catalog. Until they 
+finish their work with this version, it cannot be truncated.
+
+This module introduces [CatalogCompactionRunner](src/main/java/org/apache/ignite/internal/catalog/compaction/CatalogCompactionRunner.java) 
+component, which is responsible for periodically performing catalog compaction ensuring that
+dropped versions of the catalog are no longer needed by any component in the cluster.
+
+## Compaction restrictions
+
+1. Catalog must not be compacted by version which activation time is greater than or equal to earliest
+   active transaction in the cluster.
+2. Catalog must not be compacted by version which can be required to replay the raft log during recovery.
+3. Index building is stick with a specific catalog version. This version cannot be truncated until
+   the index build is complete.
+4. Rebalance is stick with a specific catalog version. This version cannot be truncated until the rebalance
+   is complete.
+
+## Coordinator
+
+Compaction is performed from single node called compaction coordinator.
+To simplify the selection of the coordinator, it was decided to consider it to be the same node 
+as the leader of the metastorage group.
+Therefore, when the metastorage group leader changes, the compaction coordinator also changes.
+
+The [ElectionListener](../metastorage/src/main/java/org/apache/ignite/internal/metastorage/impl/ElectionListener.java)
+interface was introduced to listen for metastore leader elections.
+
+## Triggering factors
+
+The process is initiated by one of the following events:
+
+1. `low watermark` has been changed
```

Comment:
Low watermark can be changed too frequently, right?
And the compaction process is not fast/lightweight, right?
If both answers are 'yes', then... is there any back off mechanics?

Reply:
> Low watermark can be changed too frequently, right?
it's 5 mins by default, but it's configurable :thinking: 

>  is there any back off mechanics?
there is no back off mechanics, only that the process can't start if previous iteration is not finished yet.

Reply:
Added this into [IGNITE-23059](https://issues.apache.org/jira/browse/IGNITE-23059)

Reply:
Let's just mention `low watermark` is triggered sporadically (every 5min by default).